src/codec.ts

```diff
@@ -68,17 +68,25 @@ export function stringTuplesToString (tuples: StringTuple[]): string {
 /**
  * [[str name, str addr]... ] -> [[int code, Uint8Array]... ]
  */
-export function stringTuplesToTuples (tuples: Array<string[] | string>): Tuple[] {
-  return tuples.map((tup) => {
+export function stringTuplesToTuples (stringTuples: Array<string[] | string>): { tuples: Tuple[], path: string | null } {
+  let path: string | null | undefined
+  const tuples = stringTuples.map((tup) => {
     if (!Array.isArray(tup)) {
       tup = [tup]
     }
     const proto = protoFromTuple(tup)
-    if (tup.length > 1) {
-      return [proto.code, convertToBytes(proto.code, tup[1])]
+    const tuple: Tuple = (tup.length > 1) ? [proto.code, convertToBytes(proto.code, tup[1])] : [proto.code]
+    if (path === undefined && proto.path === true) {
+      path = tuple[1] != null ? convertToString(proto.code, tuple[1]) : null
     }
-    return [proto.code]
+    return tuple
   })
+
+  if (path == null) {
+    path = null
+  }
```

Comment:
Don't get it. If value for `path` is not set, we are resetting it.

Reply:
check this implementation https://github.com/multiformats/js-multiaddr/blob/76fa6f57b842010c2f74fffdca35b9a1435dbab9/src/index.ts#L699

when there is no path in any protocols, it means `path = null` and we don't need to search for it again the next time getPath() is called 

I'll add more comments to this function referring to `DefaultMultiaddr.getPath()`

Reply:
Yes but above condition is not useful, its like setting `if path is null set it to null`. Is `path` is already `null` we don't need to set again.

Reply:
@nazarhussain  yeah it's a bit confused, I changed the condition to `if (path === undefined)` instead